src/main/kotlin/com/openlattice/data/ids/PostgresEntityKeyIdService.kt

```diff
@@ -57,193 +58,160 @@ private val entityKeyIdsSql = "SELECT * FROM ${SYNC_IDS.name} WHERE ${ENTITY_SET
 private val entityKeyIdSql = "SELECT * FROM ${SYNC_IDS.name} WHERE ${ENTITY_SET_ID.name} = ? AND ${ENTITY_ID.name} = ? "
 private val linkedEntityKeyIdsSql = "SELECT ${ID.name},${LINKING_ID.name} as $ENTITY_KEY_IDS_FIELD FROM ${IDS.name} WHERE ${ID.name} = ANY(?) AND ${LINKING_ID.name} IS NOT NULL"
 
-private val INSERT_SQL = "INSERT INTO ${IDS.name} (${ENTITY_SET_ID.name},${ID.name},${PARTITION.name}) VALUES(?,?,?)"
+private val INSERT_SQL = "INSERT INTO ${IDS.name} (${ENTITY_SET_ID.name},${ID.name},${PARTITION.name}) VALUES(?,?,?) ON CONFLICT DO NOTHING"
 private val INSERT_ID_TO_DATA_SQL = "INSERT INTO ${DATA.name} (" +
         "${ENTITY_SET_ID.name}," +
         "${ID.name}," +
         "${PARTITION.name}," +
         "${PROPERTY_TYPE_ID.name}," +
         "${VERSION.name}," +
-        "${HASH.name}) VALUES (?,?,?,?,?,?)"
-private val INSERT_SYNC_SQL = "INSERT INTO ${SYNC_IDS.name} (${ENTITY_SET_ID.name},${ENTITY_ID.name},${ID.name}) VALUES(?,?,?)"
+        "${HASH.name}) VALUES (?,?,?,?,?,?) " +
+        "ON CONFLICT DO NOTHING"
+private val INSERT_SYNC_SQL = "INSERT INTO ${SYNC_IDS.name} (${ENTITY_SET_ID.name},${ENTITY_ID.name},${ID.name}) VALUES(?,?,?) ON CONFLICT DO NOTHING"
 
 private val logger = LoggerFactory.getLogger(PostgresEntityKeyIdService::class.java)
 
 class PostgresEntityKeyIdService(
-        hazelcastClients: HazelcastClientProvider,
-        private val executor: ListeningExecutorService,
         private val hds: HikariDataSource,
         private val idGenerationService: HazelcastIdGenerationService,
         private val partitionManager: PartitionManager
 ) : EntityKeyIdService {
-    private val hazelcastInstance = hazelcastClients.getClient(HazelcastClient.IDS.name)
-    private val idRefCounts = HazelcastMap.ID_REF_COUNTS.getMap(hazelcastInstance)
-    private val idMap = HazelcastMap.ID_CACHE.getMap(hazelcastInstance)
-
     private fun genEntityKeyIds(entityIds: Set<EntityKey>): Map<EntityKey, UUID> {
         val ids = idGenerationService.getNextIds(entityIds.size)
         checkState(ids.size == entityIds.size, "Insufficient ids generated.")
         return entityIds.zip(ids).toMap()
     }
 
     private fun storeEntityKeyIdReservations(
-            entitySetId: UUID, entityKeyIds: Set<UUID>,
-            partitions: Set<Int> = partitionManager.getEntitySetPartitions(entitySetId)
+            entitySetId: UUID,
+            entityKeyIds: Set<UUID>,
+            partitions: Array<Int> = partitionManager.getEntitySetPartitions(entitySetId).toTypedArray()
```

Comment:
Did we determine that accessing partitionManager in the default value function is safe? (I vaguely remember problems with this in the distant and recent past but don't remember what the resolution was)

Reply:
it's safe as long field the service isn't annotated with ```@Timed``` or similar annotations that trigger spring proxying the class. Even then it's fine as long as member variable is declared protected (or public).

Reply:
Also, what's the reason to convert `partitions` to an array?

Reply:
I thought it was also being bound as a jdbc parameter, so I was trying to avoid unnecessary copy.  By the time I realized it was array everywhere. It's perfectly reason to switch to List<Int> instead to avoid having to versions of the function since no one is likely to save time in the way I originally imagined since this is used for look up only and never binding.

Reply:
Any chance we can use IntArray instead to keep it primitive?

Reply:
Sure